drivers/storage/portworx/deployment.go

```diff
@@ -359,6 +389,60 @@ func (p *portworx) GetStoragePodSpec(
 	return podSpec, nil
 }
 
+func (p *portworx) getCloudStorageSpec(nodes []corev1alpha1.StorageNode, cluster *corev1alpha1.StorageCluster) (*cloudstorage.Config, error) {
+	return p.generateCloudStorageSpecs(cluster, nodes)
+}
+
+func (p *portworx) createStorageNode(cluster *corev1alpha1.StorageCluster, nodeName string, cloudConfig *cloudstorage.Config) error {
+	ownerRef := metav1.NewControllerRef(cluster, controllerKind)
+
+	storageNode := &corev1alpha1.StorageNode{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            nodeName,
+			Namespace:       cluster.Namespace,
+			OwnerReferences: []metav1.OwnerReference{*ownerRef},
+			Labels:          p.GetSelectorLabels(),
+		},
+		Status: corev1alpha1.NodeStatus{},
+	}
+
+	configureStorageNodeSpec(storageNode, cloudConfig)
+	cluster.Status.StorageInstancesPerZone = cloudConfig.StorageInstancesPerZone
+
+	return k8s2.CreateOrUpdateStorageNode(p.k8sClient, storageNode, ownerRef)
```

Comment:
If we only want to create StorageNode here, we should just call `p.k8sClient.Create()` to avoid unwanted updates -
```
err := p.k8sClient(context.TODO(), storageNode)
if err != nil && !errors.IsAlreadyExists(err) {
  return err
}
return nil
```

Reply:
what kind of unwanted updates? this function is called only if 
`!storageNodeExists(nodeName, nodes) `

Reply:
or you want to split `CreateOrUpdateStorageNode` into 2 functions?

Reply:
No need to split, you just call `p.k8sClient(context.TODO(), storageNode)`, which is what `CreateOrUpdateStorageNode` does internally. `CreateOrUpdateStorageNode` also updates the object if it is present, which is not our desired behavior here.

Reply:
That's why just call create, no need to call CreateOrUpdate if you are not planning to Update anyway.